packages/core/src/odata/common/selectable/enum-field.ts

```diff
@@ -0,0 +1,82 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+/* eslint-disable max-classes-per-file */
+
+import { EdmTypeShared } from '../edm-types';
+import { EntityBase, ODataVersionOf, Constructable } from '../entity';
+import {
+  ComplexTypeField,
+  getEdmType,
+  getEntityConstructor
+} from './complex-type-field';
+import { ConstructorOrField } from './constructor-or-field';
+import { EdmTypeField, SelectableEdmTypeField } from './edm-type-field';
+
+/**
+ * Represents a property with a enum value.
+ *
+ * @typeparam EntityT - Type of the entity the field belongs to
+ */
+class EnumFieldBase<EntityT extends EntityBase> extends EdmTypeField<
+  EntityT,
+  string
+> {}
+
+/**
+ * Represents a selectable property with a enum value.
+ *
+ * @typeparam EntityT - Type of the entity the field belongs to
+ */
+export class EnumField<EntityT extends EntityBase>
+  extends EnumFieldBase<EntityT>
+  implements SelectableEdmTypeField {
+  readonly selectable: true;
+
+  edmEnumType: string;
```

Comment:
In my IDE these two are marked as unused. Are they used somewhere and if so for what?